spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/BatchMessagingMessageListenerAdapter.java

```diff
@@ -107,6 +109,28 @@ public void setBatchToRecordAdapter(BatchToRecordAdapter<K, V> batchToRecordAdap
 		this.batchToRecordAdapter = batchToRecordAdapter;
 	}
 
+	/**
+	 * Set the {@link SmartMessageConverter} to use with the batch message converter.
+	 * <p>
+	 * When a {@code SmartMessageConverter} is configured via
+	 * {@code @KafkaListener(contentTypeConverter = "...")}, this method ensures it is
+	 * properly propagated to the batch converter, which will then propagate it to the
+	 * record converter for message conversion in batch listeners.
+	 * <p>
+	 * This override does not call the parent implementation because the parent's validation
+	 * (checking {@code converterSet}) blocks setting the SmartMessageConverter after
+	 * {@code setBatchMessageConverter} has been called, which is the normal workflow for
+	 * batch listeners.
+	 * @param messageConverter the converter to set
+	 */
+	@Override
+	public void setMessagingConverter(SmartMessageConverter messageConverter) {
+		if (this.batchMessageConverter instanceof BatchMessagingMessageConverter) {
+			((BatchMessagingMessageConverter) this.batchMessageConverter)
+					.setMessagingConverter(messageConverter);
```

Comment:
You can use pattern expression variable to avoid this new line breaking and cast at all:
```
if (this.batchMessageConverter instanceof BatchMessagingMessageConverter messagingConverter) {
			messagingConverter.setMessagingConverter(messageConverter);
```

Reply:
Has not been addressed.